python/src/indexing.cpp

```diff
@@ -418,6 +424,15 @@ array mlx_get_item_nd(array src, const nb::tuple& entries) {
         out_shape.end(), src.shape().begin() + axis, src.shape().end());
 
     src = reshape(src, out_shape);
+  } else if (none_exists) {
+    std::vector<int> out_shape = src.shape();
+    int shifted = 0;
+    for (size_t i = 0; i < remaining_indices.size(); i++) {
+      if (remaining_indices[i].is_none()) {
+        out_shape.insert(out_shape.begin() + (shifted++) + i, 1);
+      }
+    }
+    src = reshape(src, out_shape);
```

Comment:
We can just enter the reshape loop above without the need for another branch 
What caused the bug is that is previously the expansion of ellipsis would also pad the end of the array with none slices (hence remaining indices would cover all dimensions for the src making the unsqueezed check be `remaining_indices.size() > src.ndim()`)

You can juts check if a unsqueeze is needed in the loop that prepares the slices and then enter the reshape branch above